### ISSUE TITLE
pyarmor.helper.repack script fails if the PYZ archive generated by PyInstaller contains subdirectories

### ISSUE THREAD

**Issue by ntamas** [dashingsoft/pyarmor#415]
When repacking a PyInstaller single-file executable where the embedded PYZ archive contains subdirectories, the repack script fails when it tries to extract the files from the embedded PYZ archive. I think the easiest way to achieve this is by trying to repack any PyInstaller-based app that depends on `bcrypt`; I get an error like this on my machine when running the repack script in debug mode:

```
[...]
190 DEBUG: toc: (1900410, 50503, 162832, 1, 'b', '_ssl.cpython-38-darwin.so')
191 DEBUG: toc: (1950913, 13171, 75312, 1, 'b', '_struct.cpython-38-darwin.so')
191 DEBUG: toc: (1964084, 1243, 49960, 1, 'b', '_uuid.cpython-38-darwin.so')
192 DEBUG: toc: (1965327, 18215, 78296, 1, 'b', 'array.cpython-38-darwin.so')
192 DEBUG: toc: (1983542, 20436, 45492, 1, 'b', 'bcrypt/_bcrypt.abi3.so')
Traceback (most recent call last):
  File "repack.py", line 336, in <module>
    main()
  File "repack.py", line 328, in main
    repacker(args.executable, args.obfpath)
  File "repack.py", line 280, in repacker
    with open(pathnm, 'wb') as f:
FileNotFoundError: [Errno 2] No such file or directory: '******/bcrypt/_bcrypt.abi3.so'
```

The issue can easily be fixed with a call to `os.makedirs(os.path.dirname(pathnm), exist_ok=True)` before trying to open the output file for writing.

**Comment by jondy** [comment]
Fixed it with commit a2620931dc2789fd32c6ee70443ea4ec59373e6b

**Comment by ntamas** [comment]
Awesome, thanks a lot for the quick response!
